juju/juju
# Integration tests for lxd profiles

## Description of change

The integration tests here are to ensure that we add the correct lxd
profile and that we correctly remove the lxd profile. This just sets up
that the profiles are correctly installed and update correctly.

More work is required to ensure via the charm that we error out if the
profile isn't applied.

## QA steps

```sh
cd tests && ./main.sh deploy
```

Should output something like:

```
➜ ./main.sh deploy

==> Checking for dependencies
==> TEST BEGIN: test_deploy (tmp.coR)
==> Checking for dependencies
====> Unable to reuse bootstrapped juju
====> Bootstrapping juju (2.8.1:lxd)
====> Bootstrapped juju (120s)
===> [ ✔ ] Success: deploy charm (194s)
===> [ ✔ ] Success: deploy lxd to container (152s)
===> [ ✔ ] Success: deploy lxd profile charm container (171s)
===> [ ✔ ] Success: deploy lxd to machine (80s)
===> [ ✔ ] Success: deploy lxd profile charm (84s)
===> [ ✔ ] Success: deploy local lxd profile charm (461s)
====> Destroying model (test-deploy-ctl)
====> Destroyed model (test-deploy-ctl)
==> TEST DONE: test_deploy (1266s)
==> Cleaning up
====> Cleaning up jujus
====> Introspection gathering
Querying @jujud-machine-0 introspection socket: /depengine
Querying @jujud-machine-0 introspection socket: /debug/pprof/goroutine?debug=1
====> Introspection gathered
====> Removing offers
====> Removed offers
====> Destroying juju (ctrl-4apfhtrp)
====> Destroyed juju (ctrl-4apfhtrp)
====> Completed cleaning up jujus

==> Test result: success
==> Tests Removed: /home/simon/go/src/github.com/juju/juju/tests/tmp.coR
==> TEST COMPLETE
```
